PC/launcher.c

```diff
@@ -1040,32 +1046,47 @@ find_terminator(char * buffer, int len, BOM *bom)
 static BOOL
 validate_version(wchar_t * p)
 {
-    BOOL result = TRUE;
-
-    if (!isdigit(*p))               /* expect major version */
-        result = FALSE;
-    else if (*++p) {                /* more to do */
-        if (*p != L'.')             /* major/minor separator */
+    /*
+    Version information should start with one of 2 or 3,
+    Optionally followed by a period and a minor version,
+    Optionally followed by a minus and one of 32 or 64.
+    Valid examples:
+      2
+      3
+      2.7
+      3.6
+      2.7-32
+      The intent is to add to the valid patterns:
+      3.10
+      3-32
+      3.6-64
+      3-64
+    */
+    BOOL result = (p != NULL) && *p;  /* Default to false if empty string or null pointer. */
+
+    while (result && isdigit(*p))   /* Require a major version */{
```

Comment:
You're not requiring a major version -- just skipping leading digits. What if the version string is ".7-32"? ISTM, you should initially set `BOOL result = p && isdigit(*p); `. Then you can skip remaining digits. 

Also, this should be calling `iswdigit`. The debug CRT raises an exception if `isdigit` is passed a value greater than 255. As things stand I think py.exe always links statically with a non-debug implementation, but it's still not a good practice to use `isdigit` with 16-bit `wchar_t` values.